tools/validators/instance_validator/instance_validator.py

```diff
@@ -105,24 +101,34 @@ def message_handler(message):
 
   print('Universe generated successfully')
 
-  parsed = dict(raw_parse)
+  parsed_entities = dict(raw_parse)
   entity_instances = {}
-  entity_names = list(parsed.keys())
+  entity_names = set(parsed_entities.keys())
   # first build all the entity instances
-  for entity_name in entity_names:
-    entity = dict(parsed[entity_name])
+  for name, entity in parsed_entities.items():
     instance = entity_instance.EntityInstance(entity,
                                               universe,
-                                              set(entity_names))
-    entity_instances[entity_name] = instance
+                                              entity_names)
+    entity_instances[name] = instance
 
   for entity_name, entity_instance in entity_instances.items():
     if not entity_instance.IsValidEntityInstance(entity_instances):
       print(entity_name, 'is not a valid instance')
       sys.exit(0)
 
   print('File passes all checks!')
+
   if pubsub_validation_set:
     print('Connecting to pubsub subscription: ', arg.subscription)
     sub = subscriber.Subscriber(arg.subscription, arg.service_account)
-    sub.Listen(message_handler)
+    timeout = arg.timeout if arg.timeout is not None else DEFAULT_TIMEOUT
```

Comment:
you can set the default timeout in the argparse during the definition of timeout

Reply:
Done